kamu-core/tests/tests/test_reset_service_impl.rs

```diff
@@ -0,0 +1,184 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::sync::Arc;
+use std::path::Path;
+
+use kamu::domain::*;
+use kamu::infra::*;
+use kamu::testing::*;
+use opendatafabric::*;
+
+
+#[test_log::test(tokio::test)]
+async fn test_reset_dataset_with_2revisions_drop_last() {
+    let tmp_dir = tempfile::tempdir().unwrap();
+    let harness = ResetTestHarness::new(tmp_dir.path());
+    let test_case = harness.a_chain_with_2_blocks().await;
+
+    let current_head = harness.get_dataset_head(&test_case.dataset_handle).await;
+    assert_eq!(test_case.hash_polling_source_block, current_head);        
+
+    harness.reset_dataset(
+        &test_case.dataset_handle, 
+        &test_case.hash_seed_block
+    ).await;
+
+    let new_head = harness.get_dataset_head(&test_case.dataset_handle).await;
+    assert_eq!(test_case.hash_seed_block, new_head);    
+
+    let summary = harness.get_dataset_summary(&test_case.dataset_handle).await;
+    assert_eq!(new_head, summary.last_block_hash);    
+}
+
+#[test_log::test(tokio::test)]
+async fn test_reset_dataset_with_2revisions_without_changes() {
+    let tmp_dir = tempfile::tempdir().unwrap();
+    let harness = ResetTestHarness::new(tmp_dir.path());
+    let test_case = harness.a_chain_with_2_blocks().await;
+
+    let current_head = harness.get_dataset_head(&test_case.dataset_handle).await;
+    assert_eq!(test_case.hash_polling_source_block, current_head);        
+
+    harness.reset_dataset(
+        &test_case.dataset_handle, 
+        &test_case.hash_polling_source_block
+    ).await;
+
+    let new_head = harness.get_dataset_head(&test_case.dataset_handle).await;
+    assert_eq!(current_head, new_head);
+
+    let summary = harness.get_dataset_summary(&test_case.dataset_handle).await;
+    assert_eq!(current_head, summary.last_block_hash);
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
```

Comment:
Might need a test for error case when resetting to a non-existing block

Reply:
Added